### PR TITLE
change the score function to prevent drag

### PR DESCRIPTION
**Pull request by steveoni**
this pr is a propose solution to fix the drag/slow issue base on this #279 #252

### CODE REVIEW

**Review by timc1 — COMMENTED** [review]
Hey @steveoni! FYI typing anything in the search input causes the site to crash.

**Comment by steveoni** [comment]
@timc1 i will test it now.

i just copied the code I updated from work and edit directly on github and push. i will update you once I fix that. FYI that's what we are using here https://www.datopian.com/

**Comment by steveoni** [comment]
@timc1 is actually a small fix. now working without crashing the page

File: package.json [hunk 1]
```diff
@@ -60,6 +60,7 @@
     "@radix-ui/react-portal": "^1.0.1",
     "command-score": "^0.1.2",
```
[the remark this answers is not in the document]
Comment (steveoni) [reply package.json]: @timc1 done

**Comment by rareyesdev** [comment]
Thank you both for following up on this issue. I was debugging this myself. I'm adding a profiler screenshot proving that `command-score` is the culprit.

✅ It seems that `fusejs` won't get stuck anymore (as the search terms grow).

@timc1 Will this fix hit `pliny` automatically? I would like to get this fix in my blog.

image

**Comment by timc1** [comment]
Hey @rareyesdev! I'll get this update merged in soon; want to look at the configs a little more prior. You will just need to install the latest version once I get it released :)

**Review by timc1 — APPROVED** [review]

File: src/useMatches.tsx [hunk 2]
```diff
@@ -74,7 +74,29 @@ export function useMatches() {
...
+    includeScore: true,
+    includeMatches: true,
+    threshold: 0.2,
+    minMatchCharLength: 1,
+    tokenize: (str) => {
+      // Example: Preserve hyphens and special characters as separate tokens
+      return str.split(/[\s\-,.!()]+/).filter(Boolean);
+    },
+  };
+  const fuse = new Fuse(filtered, fuseOptions);
```
Comment (tommoor) [inline src/useMatches.tsx:97]: Don't you want to move this inside the `useMemo` and only create a new fuse instance when `filtered` changes? It's essentially rebuilding the search index on every render

Comment (timc1) [reply src/useMatches.tsx:97]: Yah https://github.com/timc1/kbar/pull/329

**Pull request merged**